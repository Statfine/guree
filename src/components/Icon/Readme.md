图标集见 fontawesome 的 `free && brands &&  regular && solid` 分类，参见[此地址](https://fontawesome.com/icons?d=gallery&s=brands,regular,solid&m=free)
### 基本用法
使用 `name` 设置图标名称
```jsx
<Icon name="adn"></Icon>
```
### 图标大小
使用 `size` 设置图标大小
```jsx
const value = 40;
<div>
  <Icon name="adn" size="0.5em"></Icon>
  <Icon name="adn" size="18px"></Icon>  
  <Icon name="adn" size={value}></Icon>
</div>
```
### 图标颜色
使用 `fill` 设置图标颜色
```jsx
  <div>
    <Icon name="adn" fill="red"></Icon>
  </div>
```

