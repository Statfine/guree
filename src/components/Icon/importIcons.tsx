const importAll = (requireContext:any) => requireContext.keys().map(requireContext) 
try {
  const req = require.context('../../assets/svgs/', true, /\.svg$/)
  importAll(req)
} catch (e) {

}