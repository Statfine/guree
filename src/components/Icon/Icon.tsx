import React from "react";
import "./importIcons.tsx";
import "./Icon.scss";
import { scopedClass } from "@utils/index";

interface Props {
  /** 图标名称 */
  name: string;
  /** 图标大小，可以是数字或字符串，比如 10 1em 20px 等，数字默认单位是 px */
  size?: string | number;
  /** 图标颜色 */
  fill?: string;
  /** 点击图标触发 */
  onClick?: React.MouseEventHandler<SVGElement>;
}

const Icon: React.FunctionComponent<Props> = ({
  name,
  size = "1em",
  fill,
  onClick,
}: Props) => {
  const sizeValue = (size?: string | number) => {
    return typeof size === "number" ? `${size}px` : size;
  };

  return (
    <svg
      className={scopedClass("icon")}
      style={{
        width: sizeValue(size),
        height: sizeValue(size),
        fill,
      }}
      onClick={onClick}
    >
      <use xlinkHref={`#${name}`} />
    </svg>
  );
};

export default Icon;
