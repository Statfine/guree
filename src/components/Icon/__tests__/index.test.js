import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '..'
// import "../importIcons";

it('Icon exports modules correctly', () => {
  const icon = renderer
    .create(<Icon name="adn"></Icon>)
    .toJSON();
  expect(icon).toMatchSnapshot();
});
