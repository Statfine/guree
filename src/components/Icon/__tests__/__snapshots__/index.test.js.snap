// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon exports modules correctly 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="#adn"
  />
</svg>
`;
